src/nvim/search.c

```diff
@@ -998,41 +1009,37 @@ static int first_submatch(regmmatch_T *rp)
   return submatch;
 }
 
-/*
- * Highest level string search function.
- * Search for the 'count'th occurrence of pattern 'pat' in direction 'dirc'
- *    If 'dirc' is 0: use previous dir.
- *    If 'pat' is NULL or empty : use previous string.
- *    If 'options & SEARCH_REV' : go in reverse of previous dir.
- *    If 'options & SEARCH_ECHO': echo the search command and handle options
- *    If 'options & SEARCH_MSG' : may give error message
- *    If 'options & SEARCH_OPT' : interpret optional flags
- *    If 'options & SEARCH_HIS' : put search pattern in history
- *    If 'options & SEARCH_NOOF': don't add offset to position
- *    If 'options & SEARCH_MARK': set previous context mark
- *    If 'options & SEARCH_KEEP': keep previous search pattern
- *    If 'options & SEARCH_START': accept match at curpos itself
- *    If 'options & SEARCH_PEEK': check for typed char, cancel search
- *
- * Careful: If spats[0].off.line == TRUE and spats[0].off.off == 0 this
- * makes the movement linewise without moving the match position.
- *
- * Return 0 for failure, 1 for found, 2 for found and line offset added.
- */
-int do_search(
-    oparg_T         *oap,           // can be NULL
-    int dirc,                       // '/' or '?'
-    int search_delim,  // delimiter for search, e.g. '%' in s%regex%replacement
-    char_u          *pat,
-    long count,
-    int options,
-    searchit_arg_T  *sia        // optional arguments or NULL
-)
+/// Highest level string search function.
+/// Search for the 'count'th occurrence of pattern 'pat' in direction 'dirc'
+///    If 'dirc' is 0: use previous dir.
+///    If 'pat' is NULL or empty : use previous string.
+///    If 'options & SEARCH_REV' : go in reverse of previous dir.
+///    If 'options & SEARCH_ECHO': echo the search command and handle options
+///    If 'options & SEARCH_MSG' : may give error message
+///    If 'options & SEARCH_OPT' : interpret optional flags
+///    If 'options & SEARCH_HIS' : put search pattern in history
+///    If 'options & SEARCH_NOOF': don't add offset to position
+///    If 'options & SEARCH_MARK': set previous context mark
+///    If 'options & SEARCH_KEEP': keep previous search pattern
+///    If 'options & SEARCH_START': accept match at curpos itself
+///    If 'options & SEARCH_PEEK': check for typed char, cancel search
+///
+/// Careful: If spats[0].off.line == TRUE and spats[0].off.off == 0 this
+/// makes the movement linewise without moving the match position.
+///
+/// @param oap           can be NULL
+/// @param dirc          '/' or '?'
+/// @param search_delim  delimiter for search, e.g. '%' in s%regex%replacement
+/// @param sia           optional arguments or NULL
+///
+/// @return              0 for failure, 1 for found, 2 for found and line offset added.
+int do_search(oparg_T         *oap, int dirc, int search_delim, char_u          *pat, long count,
```

Comment:
I suppose re-running uncrustify will fix this?

Reply:
Nah, there are a bunch of options specifically for asterisk that I've ignored until now. I've fixed this but not sure if the fix unfixed something else.